universal-login-react/src/ui/commons/Assets.tsx

```diff
@@ -10,20 +10,40 @@ import './../styles/themes/Jarvis/assetsListThemeJarvis.sass';
 import {getIconForToken} from '../../core/utils/getIconForToken';
 import {useBalances} from '../hooks/useBalances';
 import {ThemedComponent} from './ThemedComponent';
+import DisplayTokens from './DisplayTokens';
+import Spinner from './Spinner';
+import {classForComponent} from '../utils/classFor';
+import {IToken, getErc721Tokens} from '../../core/utils/getErc721Tokens';
+
+export type assetState = 'COINS' | 'TOKENS' | 'LOADING';
 
 export interface AssetsProps {
   deployedWallet: DeployedWallet;
 }
 
 export const Assets = ({deployedWallet}: AssetsProps) => {
   const [tokenDetailsWithBalance] = useBalances(deployedWallet);
+  const [currentState, setCurrentState] = useState('COINS' as assetState);
+  const [erc721Tokens, setErc721Tokens] = useState([] as IToken[]);
+
+  const onCoinsClick = () => {
+    if (currentState === 'LOADING') setErc721Tokens([]);
+    setCurrentState('COINS');
+  };
+
+  const onTokensClick = async () => {
+    setCurrentState('LOADING');
+    setErc721Tokens(await getErc721Tokens(deployedWallet.contractAddress));
+    setCurrentState('TOKENS');
+  };
 
   return (
     <ThemedComponent name="assets">
       <div className="assets">
-        <p className="assets-title">My Assets</p>
+        <a className={`${currentState !== 'COINS' ? 'assets-title' : 'assets-disabled-title'}`} onClick={() => {onCoinsClick();}}>Coins</a>
+        <a className={`${currentState === 'COINS' ? 'assets-title' : 'assets-disabled-title'}`} onClick={async () => {await onTokensClick();}}>Tokens</a>
```

Comment:
async () => {await onTokensClick();} ??

Reply:
Why not?

Reply:
it is redundant async, we can write it: `() = > onTokensClick()`